### ISSUE TITLE
Request for a full Dark/Black Background in FeatherNotes in the page where all text is written

### ISSUE THREAD

**Issue by userfeathernote** [tsujan/FeatherNotes#129]
I am a Zim user for quite a long time. However, I recently decided to plunge head-on and give FeatherNotes a spin.

It felt significantly way more responsive and snappy...to the point I am seriously considering in fully migrating to  FeatherNotes. Never-the-less, the main thing holding me back is lack of a Black/Dark grey background in the page where all text is written (which in Zim can easily be activated). Please see the attached print-screens. 

I am imploring that you kindly consider this feature, as a white background greatly negatively affects my eyes.

P.S. : I am using Arc linux and installed Kvantum. Unfortunately,  it was only able to darken the left column shown in the print-screen). QT 5 Configuration tool was likewise not successful.

![FEATHERNOTES PRUBR-SCREEN](https://user-images.githubusercontent.com/68150020/87233561-f6389400-c3c8-11ea-81ab-65eae3c26718.png)

![ZIM BACKGROUND](https://user-images.githubusercontent.com/68150020/87233566-ff296580-c3c8-11ea-8b41-5ab4241663d7.png)

**Comment by tsujan** [comment]
No, a dark editor background is not implemented because of complications in Qt, especially in exporting to PDF. With dark styles, the background color only changes to `#ececec` to avoid a too high contrast.

Maybe later...

**Comment by userfeathernote** [comment]
Thanks. 

At this juncture at least, could you kindly consider #ececec  background color where all the text is written? It presently only has a white background.

**Comment by tsujan** [comment]
It isn't white but `#ececec`, in your screenshot too. That color is ignored when exporting to PDF/HTML.

An HTML editor shouldn't follow the widget style, IMO. If a dark bg color is supported, it should be added to the HTML code (the saved note), so that it'll be respected with HTML exporting and be persistent with all widget styles. Hence the complication...

**Comment by tsujan** [comment]
Actually, when a dark bg is supported, any background and foreground colors should also be supported, regardless of the widget style. *The saved note* should determine them, exactly as it determines the document/side-pane font and its size: the app starts with the system font, a white or `#ececec` editor background and a black editor text; then the user could change them in any way and save the document, so that whenever it's opened, those colors will be intact.

**Comment by userfeathernote** [comment]
Thanks for the additional clarification.

....I am keeping my fingers crossed to a black/dark-grey background at the earliest.

**Comment by tsujan** [comment]
I'm working on it.

There's only one problem and it exist in `QTextEdit`, which FeatherNotes uses for the rich-text support (because it's the lightest Qt widget that supports HTML): The new background/foreground colors will be applied only to new nodes; the background and foreground colors of the existing nodes could not be changed.

**Comment by tsujan** [comment]
Almost done. I added an explanation to the top of the new "document colors dialog":

![doc_colors](https://user-images.githubusercontent.com/981076/87258715-fadd7500-c4ba-11ea-81b4-4186c3d94006.png)

The colors aren't limited to black and white and the last setting is remembered inside the saved note. In this way, the user can have different background/foreground colors in different nodes of the same note. I may add a brief explanation to Help later.

The colors are also respected when the document is exported to HTML or PDF. I think they should be respected with printing too but who wants to waste a lot of ink for the background?

The main obstacle was a bad bug in `QTextDocument` but, luckily, I succeeded in making and including a simple workaround for it. So far, so good. Will make a PR after testing it for a few days.
